vortex-array/src/array/varbin/compute/compare.rs

```diff
@@ -0,0 +1,77 @@
+use arrow_array::cast::AsArray;
+use arrow_array::types::ByteArrayType;
+use arrow_array::{
+    Array as ArrowArray, BinaryArray, GenericByteArray, LargeBinaryArray, LargeStringArray, Scalar,
+    StringArray,
+};
+use arrow_cast::cast;
+use arrow_ord::cmp;
+use arrow_schema::DataType;
+use vortex_error::{vortex_bail, VortexResult};
+
+use crate::array::{ConstantArray, VarBinArray};
+use crate::arrow::FromArrowArray;
+use crate::compute::{MaybeCompareFn, Operator, SliceFn};
+use crate::{Array, IntoCanonical};
+
+impl MaybeCompareFn for VarBinArray {
+    fn maybe_compare(&self, array: &Array, operator: Operator) -> Option<VortexResult<Array>> {
+        if let Ok(rhs_const) = ConstantArray::try_from(array) {
+            Some(compare_constant(self, &rhs_const, operator))
+        } else {
+            None
+        }
+    }
+}
+
+fn compare_constant(
+    lhs: &VarBinArray,
+    rhs: &ConstantArray,
+    operator: Operator,
+) -> VortexResult<Array> {
+    let arrow_lhs = lhs.clone().into_canonical()?.into_arrow();
+    let constant = rhs.slice(0, 1)?.clone();
+    let arrow_rhs = constant.into_canonical()?.into_arrow();
+
+    match arrow_lhs.data_type() {
+        DataType::Binary => {
+            let arrow_rhs = cast(arrow_rhs.as_ref(), &DataType::Binary)?;
```

Comment:
Why is the `cast` needed here?

Reply:
I guess we can have a fallible downcast? Not clear that the types are checked anywhere else